### ISSUE TITLE
Strange 404 handling

### ISSUE THREAD

**Issue by gpopovic** [pressly/chi#164]
It appears that 404 needs to be set up for each particular group and this is annoying and wrong.

Below is the code used to reproduce the bug.

Basically route /awdad will correctly call NotFound handler.

Routes starting with /a, .. will not work(http.NotFound will be called instead of our NotFound handler)... since there is no NotFound handler defined in the router group.

Routes starting with /b,... will work because NotFound handler is defined inside router group...Is this how it should work? :(

Shouldn't NotFound be global (work regardless of router groups?)

```
        app.Router.NotFound(app.NotFound)
	app.Router.Group(func(r chi.Router) {
		r.Route("/a", func(r chi.Router) {
			r.Get("/x", func(w http.ResponseWriter, r *http.Request) {
				io.WriteString(w, "uh")
			})
		})
	})

	app.Router.Group(func(r chi.Router) {
                // Without this Not found ...all 404 urls starting with /b will not really call NotFound handler
		r.NotFound(app.NotFound)
		r.Route("/b", func(r chi.Router) {
			r.Group(func(r chi.Router) {
				r.Route("/c", func(r chi.Router) {
					r.Get("/x", func(w http.ResponseWriter, r *http.Request) {
						io.WriteString(w, "uh")
					})
				})

			})

		})
	})

	app.Router.Route("/d", func(r chi.Router) {
		r.Get("/ah", func(w http.ResponseWriter, r *http.Request) {
			io.WriteString(w, "uh")
		})
	})

```

**Comment by gpopovic** [comment]
Same goes for MethodNotAllowed

**Comment by pkieltyka** [comment]
interesting, this must be the same as https://github.com/pressly/chi/issues/169

**Comment by pkieltyka** [comment]
Fixed by 5b7406a71dbb35d4f59b0b48e68cd9f9256f5262
